scripts/mgba-lua/http-server.lua

```diff
@@ -598,35 +617,136 @@ app:post("/echo", function(req, res)
+            })
+        end
+    end
+    return regions
+end
+
 -- WebSocket route
 app:websocket("/ws", function(ws)
     console:log("WebSocket connected: " .. ws.path)
 
-    ws.onMessage = function(code)
-        local function safe_eval()
-            console:log("WebSocket eval request: " .. tostring(code))
-            local chunk = code
-            
-            -- Enhanced support for non-self-executing function inputs
-            -- Check if it's already a complete statement or needs a return prefix
-            if not code:match("^%s*(return|local|function|for|while|if|do|repeat|goto|break|::|end|%(function)") then
-                chunk = "return " .. code
-            end
+    -- Initialize memory watcher for this connection
+    memoryWatchers[ws.id] = {
+        regions = {},
+        lastData = {}
+    }
+
+    ws.onMessage = function(message)
+        local function safe_handler()
+            console:log("WebSocket message: " .. tostring(message))
             
-            local fn, err = load(chunk, "websocket-eval")
-            if not fn then
-                ws:send(HttpServer.jsonStringify({error = err or "Invalid code"}))
+            -- All messages must be in simple format now
+            local simpleMessage = parseSimpleMessage(message)
+            if not simpleMessage then
+                ws:send("eval\nerror\nInvalid message format. Use simple format: command\\ndata\\ndata...")
                 return
             end
-            local ok, result = pcall(fn)
-            if ok then
-                ws:send(HttpServer.jsonStringify({result = result}))
+                if simpleMessage.command == "watch" then
+                    -- Handle memory region watching request in simple format
+                    local regions = parseWatchRegions(simpleMessage.data)
+                    if #regions > 0 then
+                        console:log("Setting up memory watch for " .. #regions .. " regions")
+                        local watcher = memoryWatchers[ws.id]
+                        watcher.regions = regions
+                        watcher.lastData = {}
+                        
+                        -- Initialize baseline data for each region
+                        for i, region in ipairs(regions) do
+                            if region.address and region.size then
+                                local data = emu:readRange(region.address, region.size)
+                                watcher.lastData[i] = data
+                            end
+                        end
+                        
+                        -- Send unified response format: watch\nsuccess\nMessage
+                        ws:send("watch\nsuccess\nMemory watching started for " .. #regions .. " regions")
+                    else
+                        -- Send unified error format: watch\nerror\nMessage  
+                        ws:send("watch\nerror\nInvalid watch regions format")
+                    end
```

Comment:
we do want all the server responses to be in json, not in this simple format. update http-server responses to all be json and update websocket client to handle json updates and message returns. remove legacy compatibility